02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,413 @@
+Let's look at the `MATR3` example below.
+
+```{r}
+human_zebrafish_mapped_df %>%
+  dplyr::filter(human_symbol == "MATR3")
+```
+
+## Collapse zebrafish genes mapping to multiple human genes
+
+Remember that if a zebrafish Ensembl gene ID maps to multiple human symbols, the values get duplicated.
+We can collapse the multi-mapped values into a list for each Ensembl ID as to not have duplicate values in our data frame.
+
+In the next chunk, we show how we can collapse all the human gene symbols into one column where we store them all for each unique zebrafish Ensembl ID.
+
+```{r}
+collapsed_human_symbol_df <- human_zebrafish_mapped_df %>%
+  # Group by zebrafish Ensembl IDs
+  dplyr::group_by(Gene) %>%
+  # Collapse the mapped values in `human_zebrafish_mapped_df` into one column named
+  # `all_human_symbols` -- note that we will lose the `support` column in this summarizing step
+  dplyr::summarize(all_human_symbols = paste(human_symbol, collapse = ";"))
+
+head(collapsed_human_symbol_df)
+```
+
+### Write results to file
+
+Now let's write our list of human gene symbols for each unique zebrafish Ensembl ID results to file.
+
+```{r}
+readr::write_tsv(
+  collapsed_human_symbol_df,
+  file.path(
+    results_dir,
+    # Replace with a relevant output file name
+    "GSE71270_zebrafish_ensembl_to_collapsed_human_gene_symbol.tsv"
+  )
+)
+```
+
+## Collapse human genes mapping to multiple zebrafish genes
+
+Since multiple zebrafish Ensembl gene IDs map to the same human symbol, we can also collapse all Ensembl IDs that each unique human gene symbol mapped to.
+
+A known issue in ortholog mapping is how to handle this data because combining values across zebrafish transcripts using an average may result in the loss of a lot of data and will likely not be representative of the zebrafish biology.
+
+Therefore, we will use the `support` column to decide which mappings to retain.
+Let's take a look at `support`.
+
+```{r}
+head(human_zebrafish_mapped_df$support)
+```
+
+Looks like we have a variety of databases for multiple mappings, but we do have some instances of only one database reported in support of the mapping.
+As we noted earlier, an orthology prediction where more than one of the databases concur would be considered reliable.
+Therefore, where we have multi-mapped zebrafish Ensembl gene IDs, we will take the mappings with more than one database to support the assertion.
+
+```{r}
+filtered_zebrafish_ensembl_df <- human_zebrafish_mapped_df %>%
+  # Count the number of databases in the support column for each prediction
+  dplyr::mutate(n_databases = stringr::str_count(support, ",") + 1) %>%
+  # Group by human gene symbols
+  dplyr::group_by(human_symbol) %>%
+  # Now filter for the rows with more than one database in support for each human gene symbol
+  dplyr::filter(n_databases > 1)
+
+head(filtered_zebrafish_ensembl_df)
+```
```

Comment:
We might look to see before and after how many multimapped genes there are. I would expect it does not go to zero, and we should note that.

Reply:
Let me know if what I added in the [last commit](https://github.com/AlexsLemonade/refinebio-examples/pull/246/commits/2717346bbcfcc5a55f001d39d027b10b40260a28#diff-6e1a150e702a9b814aaf52aa272f45dadb468d003ccbfcdbc6be1da2ccd5cb55R381) sufficiently addresses your comment here @jashapiro.